datafusion/sql/src/parser.rs

```diff
@@ -438,12 +440,26 @@ impl<'a> DFParser<'a> {
         &self,
         expected: &str,
         found: TokenWithSpan,
-    ) -> Result<T, ParserError> {
-        parser_err!(format!("Expected {expected}, found: {found}"))
+    ) -> Result<T, DataFusionError> {
+        let sql_parser_span = found.span;
+        parser_err!(format!(
```

Comment:
this actually gives me an idea that `err!` should support diagnostics as optional param, so we can remove extra `map_err`